static/js/components/email/lib.js

```diff
@@ -0,0 +1,79 @@
+import React from 'react';
+import Grid, { Cell } from 'react-mdl/lib/Grid';
+import type { Dispatch } from 'redux';
+
+import { showDialog } from '../../actions/ui';
+import {
+  startEmailEdit,
+  sendCourseTeamMail,
+  sendSearchResultMail
+} from '../../actions/email';
+import {
+  EMAIL_COMPOSITION_DIALOG,
+  COURSE_EMAIL_TYPE,
+  SEARCH_EMAIL_TYPE
+} from './constants';
+import type { Course } from '../../flow/programTypes';
+import { EmailProperties } from '../../flow/emailTypes';
+
+const courseTeamEmailProperties: EmailProperties = {
+  title: 'Contact the Course Team',
+
+  renderSubheading: (subheading: string) => (
+    <div className="subheading-section">
+      <Grid noSpacing={true}>
+        <Cell col={1} align={"middle"} className="subheading-to">TO:</Cell>
+        <Cell col={11}><h5 className="subheading rounded">{ subheading }</h5></Cell>
+      </Grid>
+    </div>
+  ),
+
+  emailOpenAction: (dispatch: Dispatch, course: Course) => {
+    dispatch(
+      startEmailEdit({
+        type: COURSE_EMAIL_TYPE,
+        params: {courseId: course.id},
+        subheading: `${course.title} Course Team`
+      })
+    );
+    dispatch(showDialog(EMAIL_COMPOSITION_DIALOG));
+  },
+
+  emailSendAction: (emailState) => (
+    sendCourseTeamMail(
+      emailState.inputs.subject || '',
+      emailState.inputs.body || '',
+      emailState.params.courseId
+    )
+  )
+};
+
+const searchResultEmailProperties: EmailProperties = {
+  title: 'New Email',
+
+  emailOpenAction: (dispatch: Dispatch, searchkit: Object) => {
+    dispatch(
+      startEmailEdit(
+        {
+          type: SEARCH_EMAIL_TYPE,
+          params: {searchkit: searchkit},
+          subheading: `${searchkit.getHitsCount() || 0} recipients selected`
+        }
+      )
+    );
+    dispatch(showDialog(EMAIL_COMPOSITION_DIALOG));
+  },
+
+  emailSendAction: (emailState) => (
+    sendSearchResultMail(
+      emailState.inputs.subject || '',
+      emailState.inputs.body || '',
+      emailState.params.searchkit.buildQuery().query
+    )
+  )
+};
```

Comment:
there are a few reasons that i separated the functionality into 3 files. i'll try to sort out my thoughts and summarize